### DISCUSSION TITLE
Shiny on Visual Studio 2022 Problem

### DISCUSSION THREAD

**Discussion by Seoxsc** [shinyorg/shiny#873] [Q&A] [answered]
Greetings, I have installed Shiny 2.4.1.2971 via NUGET in Visual Studio 2022, but it doesn't work:

public override void ConfigureServices(IServiceCollection services, IPlatform platform)
        {
            services.UseJobs(true);
            services.UseNotifications(); // adding notifications for some job fun
        }

Error: IServiceCollection does not contain a definition for UseJobs
Error: IServiceCollection does not contain a definition for UseNotifications

Am I doing something wrong?

**Comment by aritchie** [comment, marked as answer]
You have to install Shiny.Jobs & Shiny.Notifications - Shiny doesn't include all of the packages.

**Comment by Seoxsc** [comment]
Thanks for your answer, now it's working (sort of) I'm trying to follow the
documentation at https://shinylib.net/jobs/getting-started (The JOBs part)
but in the example it says CrossJobs.Current.Schedule(job); the current
version of the library doesn't have  CrossJobs.
I'm trying to implement a simple job in my app (query for update to a
webservice every 10 minutes), I'm stuck.  Could you point me to a simple
example to achieve this? Thanks in advance for your help.

**Comment by aritchie** [comment]
The documentation is not up to date because I simply haven't bothered to keep up with them.  You should also note that jobs aren't guaranteed to run on a scheduled time.  They are periodic.

There are samples located at https://github.com/shinyorg/samples as listed on the readme of this repo. 

Please note that this is not a product or a general support forum.  It is recommended you look through the source code

**Comment by Seoxsc** [comment]
Allan, thanks for your help, I managed to create the jobs and the app is
working great.  Great Job with the Plugin.

**Comment by Seoxsc** [comment]
Greetings Allan, sorry to bother you, one last question, is it mandatory to
put a Task.Delay inside the *Task Run* of the job?.  Thanks in advance for
your answer.

**Reply by aritchie** [reply]
No

**Comment by Seoxsc** [comment]
Hello Allan, sorry to bother you again, the project works great with
ANDROID (jobs run periodically), but in IOS the job never starts (not even
once), no errors, the Stratup.cs loads fine:
AppDelegate.cs:
[assembly: ShinyApplication(
    ShinyStartupTypeName = "ASPAI.Startup",
    XamarinFormsAppTypeName = "ASPAI.App"
)]
Testing in IPhoneSimulator

Everything is configured at Info.plist
<https://github.com/shinyorg/samples/blob/main/Jobs/Sample.iOS/Info.plist>
 and Entitlements.plist
<https://github.com/shinyorg/samples/blob/main/Jobs/Sample.iOS/Entitlements.plist>
according to
the examples at github. Is there something I'm mising on IOS, thanks in
advance for your help.

**Reply by aritchie** [reply]
This has been answered: https://github.com/shinyorg/shiny/discussions/888
